ArcMovies/Extensions/UIImageExtension.swift

```diff
@@ -0,0 +1,79 @@
+//
+//  UIImageExtension.swift
+//  ArcMovies
+//
+//  Created by Yuri Saboia Felix Frota on 20/01/18.
+//  Copyright © 2018 YuriFrota. All rights reserved.
+//
+
+import UIKit
+import Nuke
+
+extension UIImageView {
+    func loadImage(withURL url: URL?, defaultImage: UIImage = #imageLiteral(resourceName: "NoPoster")) {
+        let activityIndicator = UIActivityIndicatorView()
+        activityIndicator.setupOn(view: self)
+        activityIndicator.startAnimating()
+        Manager.shared.cancelRequest(for: self)
+        self.image = nil
+        guard let imageURL = url else {
+            activityIndicator.stopAnimating()
+            self.image = defaultImage
+            return
+        }
+        Manager.shared.loadImage(with: imageURL) { (images) in
+            activityIndicator.stopAnimating()
+            guard let image = images.value else {
+                self.image = defaultImage
+                return
+            }
+            self.image = image
```

Comment:
Yes, I forgot to "weakfy" self, this could lead to a retain cycle because is keeping a strong reference to the ImageView. The correct approach was to have:
 `Manager.shared.loadImage(with: imageURL) { [weak self] (images) in`
And use a guard let inside to guarantee the presence of the object being handled.